File: Java/LeetCode/Reverse Degree of a String.java
// https://leetcode.com/problems/reverse-degree-of-a-string/

class Solution {
  
    public int reverseDegree(String s) {
        int reverseDegree = 0;

        for (int i = 0; i < s.length(); ++i) {
            reverseDegree += (26 - (s.charAt(i) - 'a')) * (i+1);
        }

        return reverseDegree;
    }
  
}

File: Kotlin/LeetCode/Reverse Degree of a String.kt
// https://leetcode.com/problems/reverse-degree-of-a-string/

class Solution {

    fun reverseDegree(s: String): Int {
        var reverseDegree = 0;

        for (i in 0 until s.length) {
            reverseDegree += (26 - (s[i] - 'a')) * (i+1);
        }

        return reverseDegree;
    }

}
